### PR TITLE
Add isOptional flag

### PR DESCRIPTION
**Pull request by konstyefremov**

### CODE REVIEW

File: HappyTravel.ConsulKeyValueClient.ConfigurationProvider/ConfigurationProvider.cs [hunk 1]
```diff
@@ -48,9 +50,41 @@ private static HttpClient CreateHttpClient(string token)
...
+                    using var request = new HttpRequestMessage(HttpMethod.Get, new Uri(url, string.Empty));
+                    using var response = _httpClient.SendAsync(request)
+                        .ConfigureAwait(false)
+                        .GetAwaiter()
+                        .GetResult();
+                    
+                    if (response.IsSuccessStatusCode) 
+                        return true;
+                }
+                catch (Exception)
```
Comment (mingazhev) [inline HappyTravel.ConsulKeyValueClient.ConfigurationProvider/ConfigurationProvider.cs:68]: `(Exception)` can be omitted there

File: HappyTravel.ConsulKeyValueClient.ConfigurationProvider/ConfigurationProvider.cs [hunk 2]
```diff
@@ -48,9 +50,41 @@ private static HttpClient CreateHttpClient(string token)
...
+                {
+                    // any error means that consul is not available
```
Comment (mingazhev) [inline HappyTravel.ConsulKeyValueClient.ConfigurationProvider/ConfigurationProvider.cs:70]: Comment should start from a capital letter

File: HappyTravel.ConsulKeyValueClient.IntegrationConsole/Program.cs [hunk 3]
```diff
@@ -21,23 +21,11 @@ private static void Main(string[] _)
...
-                    services.AddConsulDiplomatProvider(ConfigFactory.FromEnvironment());
-                }).UseConsoleLifetime();
-
-            var host = builder.Build();
+            var host = new HostBuilder().UseConsoleLifetime().Build();
 
             // Testing Diplomat as a configuration provider
             var config = new ConfigurationBuilder()
-                .AddConsulKeyValueClient(new List<Uri> {new(consulUrl)}, "Development",consulPath, consulToken)
+                .AddConsulKeyValueClient(consulUrl, consulPath, consulToken, "Development", isOptional: false)
```
Comment (mingazhev) [inline HappyTravel.ConsulKeyValueClient.IntegrationConsole/Program.cs:28]: Why not `AddConsul`?

**Review by mingazhev — COMMENTED** [review]
Additional comments:
1. Rename LoadAsync to something like InitialLoad or remove the method at all. Calling the base method will be more clear if called from the overriden method
2. I suggest renaming `ExecuteQueryAsync` to `FetchConfiguration` or something similar (for more readability)

File: HappyTravel.ConsulKeyValueClient.ConfigurationProvider/ConfigurationProvider.cs [hunk 4]
```diff
@@ -19,6 +19,7 @@ public class ConsulKeyValueClientConfigurationProvider : JsonStreamConfiguration
             List<Uri> consulUrls,
             string key,
```
Comment (mingazhev) [inline HappyTravel.ConsulKeyValueClient.ConfigurationProvider/ConfigurationProvider.cs:20]: `key` may be renamed to `path`, it is not key now as for me

File: HappyTravel.ConsulKeyValueClient.ConfigurationProvider/ConfigurationProvider.cs [hunk 5]
```diff
@@ -48,9 +50,41 @@ private static HttpClient CreateHttpClient(string token)
         }
         
         
+        private bool IsConsulAvailable()
+        {
+            foreach (var url in _consulUrls)
+            {
+                try
+                {
+                    using var request = new HttpRequestMessage(HttpMethod.Get, new Uri(url, string.Empty));
```
Comment (mingazhev) [inline HappyTravel.ConsulKeyValueClient.ConfigurationProvider/ConfigurationProvider.cs:59]: Why this overload and not `new Uri(url, UrlKind.Absolute)`?

**Review by mingazhev — COMMENTED** [review]
And please if you'll have time, update documentation accordingly

File: HappyTravel.ConsulKeyValueClient.ConfigurationProvider/ConfigurationProvider.cs [hunk 6]
```diff
@@ -30,6 +31,7 @@ int delayOnFailureInSeconds
                 throw new ArgumentOutOfRangeException(nameof(consulUrls));
 
             _httpClient = CreateHttpClient(token);
+            _isOptional = isOptional;
```
Comment (kirillta) [inline HappyTravel.ConsulKeyValueClient.ConfigurationProvider/ConfigurationProvider.cs:34]: This is against our guidelines, but canonical property name for providers is `optional`

**Review by kirillta — APPROVED** [review]

**Review by mingazhev — APPROVED** [review]

**Review by mingazhev — APPROVED** [review]

**Pull request merged**